### PR TITLE
Clean data

### LINKED ISSUE #600: The Big Data Migration
1) Blow up categories that are ridiculous
2) Blow up everyone's investments
3) Reset everyone's reps to some higher number (say, 30)
4) Delete all transactions involving categories that have been deleted

Question: What should we do about the transactions for existing categories that are no longer valid? We don't want an entirely empty feed, but mocked data also kinda sucks. This can be debated over the course of the day.

### PR DESCRIPTION
**Pull request by ritmatter**
Fixes #600 

Testing: Not unit tested (I realized it's sort of silly to spend time checking error handling on these functions.
-Validated by checking the site and using mongo queries on a local copy of the production databse

### CODE REVIEW

File: bin/cleanAlphaData.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+if (process.env.NODE_ENV === 'production') {
+  winston.log('info', 'Starting to clean alpha data in production environment');
+  var mongoURL = db.production_url;
+} else {
+  winston.log('info', 'Starting to clean alpha data in development environment');
+  var mongoURL = db.development_url;
+}
+mongoose.connect(mongoURL);
+mongoose.connection.on('connected', function() {
+  utils.removeInappropriateCategories(function(errs) {
```
Comment (ritmatter) [inline bin/cleanAlphaData.js]: Remove all the inappropriate categories.
Remove all of the inappropriate users.
Reset the investors, experts, and reps for every category.

File: bin/utils.js [hunk 2]
```diff
@@ -36,6 +36,142 @@ var utils = {
...
 
+  removeInappropriateUsers: function(cb) {
+    var inappropriateUsernames = [
+      'robert frost',
+      'Noah Reichblum',
+      'Barack Obama',
+    ];
+
+    // Remove all the users with inappropriate names
+    User.remove({ 'username': { $in: inappropriateUsernames }}, function(err) {
```
Comment (ritmatter) [inline bin/utils.js]: Remove the users.
Remove every transaction to or from that user.

File: bin/utils.js [hunk 3]
```diff
@@ -36,6 +36,142 @@ var utils = {
...
+      'raptor jesus',
+      'reps',
+      'sex',
+      'spelling',
+      'television',
+      'will baird',
+    ];
+
+    // Remove all the categories with inappropriate names
+    Category.remove({ 'name': { $in: inappropriateNames }}, function(err) {
```
Comment (ritmatter) [inline bin/utils.js]: Remove the category.

File: bin/utils.js [hunk 4]
```diff
@@ -36,6 +36,142 @@ var utils = {
...
+      if (err) {
+        winston.log('error', 'Error removing inappropriate categories: %s', err.toString());
+        cb(err);
+      } else {
+        winston.log('info', 'Successfully removed inappropriate category docs.');
+
+        // Remove all of the transactions for those inappropriate categories
+        Transaction.remove({ 'category': { $in: inappropriateNames }}, function(err) {
```
Comment (ritmatter) [inline bin/utils.js]: Remove transactions that involve that category.

File: bin/utils.js [hunk 5]
```diff
@@ -36,6 +36,142 @@ var utils = {
...
+                  user = users[i];
+                  newCategories = [];
+                  newPortfolio = [];
+
+                  // Reset every user's investors
+                  // Give every early expert 10 reps
+                  for (var j = 0; j < user.categories.length; j++) {
+
+                    // Only include categories that are not being destroyed
+                    if (inappropriateNames.indexOf(user.categories[j].name) <= -1) {
```
Comment (ritmatter) [inline bin/utils.js]: Reset every users' categories and portfolio

File: bin/utils.js [hunk 6]
```diff
@@ -214,29 +352,39 @@ var utils = {
...
                   }
                 }
+                for (var p = 0; p < user.portfolio.length; p++) {
+                  if (user.portfolio[p].category === category.name) {
+                    investors++;
+                  }
+                }
               }
               category.reps = Math.floor(totalReps * 100)/100;
+              category.experts = experts;
```
Comment (ritmatter) [inline bin/utils.js]: Reset the counts of investors and experts.

**Comment by krazemon** [comment]
LGTM

**Pull request merged**